app/master/cluster_master.py

```diff
@@ -179,7 +179,7 @@ def _disconnect_slave(self, slave):
         """
         # Mark slave dead. We do not remove it from the list of all slaves. We also do not remove it from idle_slaves;
         # that will happen during slave allocation.
-        slave.is_alive = False
+        slave.set_is_alive(False)
```

Comment:
Crazy thought,  what if we used three states: ALIVE, ZOMBIE, DEAD.

ALIVE -> marked as alive, if not use cached then must be reachable via the API
ZOMBIE -> marked as not alive, but has a responsive API
DEAD -> marked as not alive, and if not use cached, the must not be reachable via the API.

For the time being, ZOMBIE ~= DEAD, but it helps differentiate whether we manually disconnected it (ZOMBIE), or the API is unresponsive (DEAD).